src/modules/shipment/index.ts

```diff
@@ -0,0 +1,59 @@
+import api from "@/api";
+import { hasError } from "@/util";
+import { Response, Shipment } from "@/types";
+import { shipmentTransformRule } from "@/mappings/shipment";
+import { transform } from "node-json-transform";
+
+async function fetchShipments(payload: any): Promise <Shipment[] | Response> {
+  let response = {} as Shipment[] | Response
+
+  const query = {
+    "inputFields": {
+      "destinationFacilityId": payload.facilityId,
+      "statusId": "PURCH_SHIP_SHIPPED",
```

Comment:
We are allowed to fetch only shipped and incoming shipments here. I would suggest making them configurable along with the following improvements:

We should define shipment status enumerations for apps (created, shipped...) and perform mapping in the package. This way package could easily take care of the different status values of different backend servers. The app query will be static for different backends having a defined status and the package does the magic. 



For incoming/outgoing shipment conditions, I would suggest 2 possible solutions:
1. We should have separate methods for incoming and outgoing shipments.
2. We should have a payload parameter, that identifies if the query is for incoming/outgoing shipment. If we consider handling with payload, we need to add these conditions to the query only when we receive certain parameters with payload.

For example, we identify a field shipmentType with values incoming/outgoing. If we do not receive any value we fetch all shipments else add
```
      "shipmentTypeId_fld0_value": "INCOMING_SHIPMENT",
      "shipmentTypeId_fld0_op": "equals",
      "shipmentTypeId_fld0_grp": "1",
      "parentTypeId_fld0_value": "INCOMING_SHIPMENT",
      "parentTypeId_fld0_op": "equals",
      "parentTypeId_fld0_grp": "2",
```
This way we are abstracting applications from server details. The app needs incoming shipments, different backends may have different data models. If we define these conditions in apps, we may diverge from the bodiless approach and have to perform workarounds when switching between backends.

Reply:
As discussed with @adityasharma7 Sir and @dt2patel Sir, we will define two separate methods for incoming and outgoing shipments.

Also will create enums for shipment type and shipment status in oms package.